### PR TITLE
web/user: refactor LibraryPage for testing, add CTA

### LINKED ISSUE #5171: UX: add an actionable button to "No Applications" page
For new installations that do not yet have any applications created, users will see a "No Applications Available" message on the Applications page of the user interface. But it is not immediately obvious how to fix the problem... can be frustrating.

Let's add a button there that takes them to the page in the UI to create an app, or to the docs on how to...

Screenshot with proposal is attached. Also, see how the Policy Binding page has a CTA button; let's do similar.

Screenshot 2023-03-09 at 8 29 59 AM

### PR DESCRIPTION
**Pull request by kensternberg-authentik**
## Details

-   Resolves #5171

## Changes

web: revise LibraryPage, add CTA

### Summary

TL;DR:

- Separated LibraryPage into a bunch of different, independent parts, none of which require Authentik running to be testable or viewable.
- This made adding the “Add an Application” CTA easier.
- This sets the stage for unit and view testing of the UI

#### Refactored

This commit revises the LibraryPage, devolving it into a couple of independent components that have to asynchronous dependencies, with a single asynchronous master:

- LibraryPage: Loads the UIConfig, UserConfig, and CoreApi, and once those are loaded, launches the LibraryPageImpl.
- LibraryPageImpl: the ListView of applications available, and updates the ListView according to search criteria it receives via an event listener.
  - LibraryPageImpl.css: The stylesheet. Put here because it’s visual clutter.
  - LibraryPageImpl.utils: defines static functions used to filter the view. Here because, again, it would otherwise be visual clutter of the LibraryPageImpl.
- ApplicationEmptyState: Shows the “You have no applications” and, if the user is a superuser, the “Add an application” button.
- ApplicationSearch: Contains the Fuse implementation and, as the search result is updated, sends the selected and filtered app list to the LibraryPage via an event. Also controls the “Choose an application by pressing Enter” event.
- ApplicationList: Displays the list of applications.

All of these components are *responsive* to changes in the Apps collection via the LibraryPage itself, but none of them invoke the Apps collection, UIConfig, and CoreApi directly, so it should be possible to create Storybook implementations that view the LibraryPageImpl itself without having to have an instance of Authentik running.

#### Extension and Bugfix

If the user is a superuser, the “You have no applications” panel now shows the “Add an Application” button and a link to the documentation on how to add an application.

### New Features

-   Adds a "Add an Application" to the LibraryView if there are no applications and the user is an administrator.

### Breaking Changes

No breaking changes.

## Checklist

-   [X] Local tests pass (`ak test authentik/`)
-   [X] The code has been formatted (`make lint-fix`)

If an API change has been made

-   No API  Changes were made.

If changes to the frontend have been made

-   [X] The code has been formatted (`make web`)
-   [X] The translation files have been updated (`make i18n-extract`)

No documentation changes were made.

### CODE REVIEW

**Review by BeryJu — COMMENTED** [review]
A bunch of nitpicks and a bunch of opinions/questions, already looking very good and should be a lot easier to test

File: web/src/user/LibraryPage/types.ts [hunk 1]
```diff
@@ -0,0 +1,12 @@
+import type { LayoutType } from "@goauthentik/common/ui/config";
+
+import type { Application } from "@goauthentik/api";
+
+export type AppGroupEntry = [string, Application[]];
+export type AppGroupList = AppGroupEntry[];
+
+export type PageUIConfig = {
```
[the remark this answers is not in the document]
Comment (BeryJu) [reply web/src/user/LibraryPage/types.ts:8]: Yeah thats a fair point, allthough maybe it would be better to go the other way around and compose `UIConfig` of smaller interfaces, and we don't have to add things in two places, but that can be a later PR if we want to do that

**Comment by tanberry** [comment]
Thanks @Ken for adding  CTA here! One request: can you change the wording about the Docs to "Refer to documentation"?

**Review by BeryJu — APPROVED** [review]
